src/storage/engines/FirebaseStorageEngine.ts

```diff
@@ -614,12 +850,29 @@ export class FirebaseStorageEngine extends StorageEngine {
   async _addDirectoryNameToMetadata(directoryName: string) {
     try {
       const metadataDoc = doc(this.firestore, 'metadata', 'collections');
-      await setDoc(metadataDoc, { [directoryName]: true }, { merge: true });
+      await setDoc(
+        metadataDoc,
+        { [directoryName]: { enabled: true, name: directoryName } },
+        { merge: true },
+      );
     } catch (error) {
       console.error('Error adding collection to metadata:', error);
     }
   }
 
+  async renameSnapshot(directoryName: string, newName: string) {
+    try {
+      const metadataDoc = doc(this.firestore, 'metadata', 'collections');
+      await setDoc(
+        metadataDoc,
+        { [directoryName]: { enabled: true, name: newName } },
+        { merge: true },
+      );
+    } catch (error) {
+      console.error('Error renaming collection in metadata', error);
```

Comment:
Should this be presented to the frontend with the new error handling you have?

Reply:
Might as well. If you like how I handled the previous one, I can do the same for this.